provider/datagen/src/driver.rs

```diff
@@ -146,105 +137,49 @@ impl fmt::Display for LocaleWithExpansion {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Default)]
 #[non_exhaustive]
-pub struct LocalesWithFallback {
+pub struct FallbackOptions {
     pub runtime_fallback_location: Option<RuntimeFallbackLocation>,
     pub deduplication_strategy: Option<DeduplicationStrategy>,
-    /// private: set via constructor
-    locales: HashSet<LocaleWithExpansion>,
+    pub und_inclusion: Option<UndInclusion>,
```

Comment:
Why are you special-casing `und`? It will either be included due to fallback, or the user can include it through `@und`.

Reply:
Good question; I wanted to make it easier to exclude `und`, which is desirable when building language packs. (Frank was having this problem building a language pack for Chrome.) Strictly speaking, this can be modeled as a `LocaleFamily` variant, such as "ancestors and descendants except for und". However, I don't see a super clean way to handle it. The options would be:

1. `--locales en-001` implies `und`, `en`, ... and the user needs to specify `--without-und` to disable `und` (what this changelist implies)
    - Pros: Puts the special case of `und` as its own flag for users who need this behavior
    - Cons: Entanglement of options
2. `--locales en-001` implies `en`, ..., (remove `und` from LocaleFamily with_ancestors) and `--with-und` (default behavior) includes `und`, and the user can specify `--without-und` to disable the inclusion of `und`
    - Pros: A cleaner model with less conflict between options
    - Cons: Still an extra flag to imply this behavior
3. `--locales en-001` implies `und`, `en`, ..., and a new LocaleFamily type `_en-001` implies `en`, ...
    - Pros: Clean model with all locale resolution handling in the same mechanism
    - Cons: Every locale needs to be prefixed with `_` in order to avoid including `und`. Need to also invent a syntax for the "ancestors, no descendants, no und" mode.
4. `--locales en-001` implies `und`, `en`, ..., and a new LocaleFamily type `!und` removes `und` from the selection
    - Pros: All locale resolution happens in the same option. New include/exclude syntax could be used to model other things.
    - Cons: Include/exclude syntax could be error-prone. We now have an ordered list of rules instead of an unordered set of families.

Do you see anything else?

Reply:
I was expecting we'd do something like 3. Including whole locale families but excluding `und` is a niche (and risky) enough behaviour that introducing a new locale family symbol shouldn't be the main problem. If the main use case is language packs, there'd usually only be a single und-less family per invocation anyway.

I don't understand why we need a "ancestors, no descendants, no und" mode. `und` is an ancestor, and it's not special. If you're building `en` language packs, you don't want to duplicate `en-001` the same way you don't want to duplicate `und`.

Reply:
Will do in a follow-up